lib/persisted-model.js

```diff
@@ -1527,6 +1546,107 @@ module.exports = function(registry) {
     });
   };
 
+  /**
+   * Subscribe to changes of this model.
+   *
+   * @param {Object} options
+   * @param {Object} options.where Only changes to models matching this where filter will be included in the `ChangeStream`.
+   * @callback {Function} callback
+   * @param {Error} err
+   * @param {ChangeStream} changes
+   */
+
+  PersistedModel.subscribe = function(options, cb) {
+    if(typeof options === 'function') {
+      cb = options;
+      options = undefined;
+    }
+
+    var idName = this.getIdName();
+    var Model = this;
+    var changes = new PassThrough({objectMode: true});
+    var writeable = true;
+    var ref = {changes: changes};
+
+    changes.destroy = function() {
+      changes.removeAllListeners('error');
+      changes.removeAllListeners('end');
+      writeable = false;
+      delete ref.changes;
+    };
+
+    changes.on('error', function() {
+      writeable = false;
+    });
+    changes.on('end', function() {
+      writeable = false;
+    });
+
+    process.nextTick(function() {
+      cb(null, changes);
+    });
+
+    Model.observe('before save', createChangeHandler('save', true));
+    Model.observe('before delete', createChangeHandler('delete', true));
```

Comment:
@bajtos 

> When running on the server, the latency of database calls is usually negligible, thus an "optimistic" update does provide much value IMO.

The purpose of an optimistic update is to avoid latency entirely. We don't have to wait for the operation to be acknowledged by the database (or rest api, other datasource) since we are only telling the client about a likely change.

I think we are disagreeing on what latency is acceptable. IMO any IO is going to add significant latency. Ideally we would emit the optimistic change right after it is sanitized. Ideally it would be emitted before timely ACL checks, but this introduces obvious vulnerabilities.

In order to move this forward, I am OK with removing this code, as it can be added later as an optimization.